### ISSUE TITLE
Unterminated String Constant

### ISSUE THREAD

**Issue by blainerobertson** [validator/validator#618]
Validator reports an error:
Bad value (?=^.{8,}$)(?=.*\d)(?=.*\W+)(?![.\n])(?=.*[A-Z])(?=.*[a-z]).*$ for attribute pattern on element input: Unterminated string constant.
Entire markup is: 
```
<input type="password" name="clientPassword" id="clientPassword"  
    pattern="(?=^.{8,}$)(?=.*\d)(?=.*\W+)(?![.\n])(?=.*[A-Z])(?=.*[a-z]).*$" required>
```

The same markup validated last week, now reports an error with no change to the markup. The pattern contains no spaces, is not multi-line and is fully enclosed in quotes.

**Comment by sideshowbarker** [comment]
Thanks very much for taking time to report this. It’s a bug but it’s not yet clear to me what the cause is. I’ll investigate.

> The same markup validated last week, now reports an error with no change to the markup.

Yeah, that’s because of a recent change I made on the backend. Seems it needs some further refinement.

**Comment by blainerobertson** [comment]
Thanks so much!

Blaine Robertson, Ed.D.
BYU-Idaho CIT Department
Email: robertsonb@byui.edu<mailto:robertsonb@byui.edu>
Voice: (208) 496-3775
Office: STC 320N
Appointments: https://calendly.com/blainer

There is no royal road to … knowledge. We must earn our achievements. To work hard and to try earnestly and to deserve good things to happen is worth the effort and worth the wait. And they will happen, often sooner than we think. But it won’t be easy and it won’t be convenient” (Euclid of Alexandria)





On Mar 2, 2018, at 5:05 PM, Michael[tm] Smith <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #618<https://github.com/validator/validator/issues/618> via d8469e3<https://github.com/validator/validator/commit/d8469e30c8743bb10307c4a455e27c5516d07e0d>.

**Comment by sideshowbarker** [comment]
> Thanks so much!

Thank you — I very much appreciate that you took time to report it. It turned out to be a pretty silly bug, so I’m glad we got it fixed before it caused problems for more people.
